eval.go

```diff
@@ -485,7 +477,15 @@ func (d *indexData) gatherMatches(nextDoc uint32, mt matchTree, known map[matchT
 				res = append(res, c)
 				continue
 			}
+
 			last := res[len(res)-1]
+
+			// Never merge filename and content matches
+			if last.fileName != c.fileName {
+				res = append(res, c)
+				continue
+			}
```

Comment:
Is it worth factoring this out or maybe moving the check for "merge" inside the loop?

Reply:
Good idea, I combined the two loops and it looks nicer